tests/e2e/txgen/txgen.go

```diff
@@ -195,6 +209,11 @@ func Generate(ctx context.Context, rtc client.RuntimeClient, rng *rand.Rand, acc
 						atomic.AddUint64(&okCount, 1)
 					}
 				}
+
+				// Perform an invariants check.
+				if err := CheckInvariants(ctx, rtc); err != nil {
+					atomic.AddUint64(&invErrCount, 1)
```

Comment:
An invariant check failing should halt the test with an error immediately.

Reply:
Fixed! :)